PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx

```diff
@@ -0,0 +1,321 @@
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file candidateCreatorSigmac0plusplus.cxx
+/// \brief Σc0,++ → Λc+(→pK-π+) π-,+ candidate builder
+/// \note Λc± candidates selected from the HFLcCandidateSelector.cxx
+///
+/// \author Mattia Faggin <mfaggin@cern.ch>, University and INFN PADOVA
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Common/Core/TrackSelection.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod::hf_cand_3prong;
+
+struct HfCandidateCreatorSigmac0plusplus {
+
+  /// Table with Σc0,++ info
+  Produces<aod::HfCandScBase> rowScCandidates;
+
+  /// Selection of candidates Λc+
+  Configurable<int> selectionFlagLc{"selectionFlagLc", 1, "Selection Flag for Lc"};
+  Configurable<double> yCandLcMax{"yCandLcMax", -1., "max. candLc. Lc rapidity"};
+  Configurable<double> mPKPiCandLcMax{"mPKPiCandLcMax", 0.03, "max. spread (abs. value) between PDG(Lc) and Minv(pKpi)"};
+  Configurable<double> mPiKPCandLcMax{"mPiKPCandLcMax", 0.03, "max. spread (abs. value) between PDG(Lc) and Minv(piKp)"};
+
+  /// Selections on candidate soft π-,+
+  Configurable<float> softPiEtaMax{"softPiEtaMax", 0.9f, "Soft pion max value for pseudorapidity (abs vale)"};
+  Configurable<int> softPiItsHitMap{"softPiItsHitMap", 127, "Soft pion ITS hitmap"};
+  Configurable<int> softPiItsHitsMin{"softPiItsHitsMin", 1, "Minimum number of ITS layers crossed by the soft pion among those in \"softPiItsHitMap\""};
+  Configurable<float> softPiDcaXYMax{"softPiDcaXYMax", 0.065, "Soft pion max dcaXY (cm)"};
+  Configurable<float> softPiDcaZMax{"softPiDcaZMax", 0.065, "Soft pion max dcaZ (cm)"};
+
+  /// Filter the candidate Λc+ used for the Σc0,++ creation
+  Filter filterSelectCandidateLc = (aod::hf_sel_candidate_lc::isSelLcToPKPi >= selectionFlagLc || aod::hf_sel_candidate_lc::isSelLcToPiKP >= selectionFlagLc);
+
+  /// Cut selection object for soft π-,+
+  TrackSelection softPiCuts;
+
+  using TracksSigmac = soa::Join<aod::FullTracks, aod::TracksDCA>;
+
+  /// @brief init function, to define the soft pion selections and histograms
+  /// @param
+  void init(InitContext&)
+  {
+
+    ////////////////////////////////////////
+    /// set the selections for soft pion ///
+    ////////////////////////////////////////
+    // softPiCuts.SetPtRange(0.001, 1000.); // pt
+    softPiCuts.SetEtaRange(-softPiEtaMax, softPiEtaMax); // eta
+    softPiCuts.SetMaxDcaXY(softPiDcaXYMax);              // dcaXY
+    softPiCuts.SetMaxDcaZ(softPiDcaZMax);                // dcaZ
+    // ITS hitmap
+    std::set<uint8_t> set_softPiItsHitMap; // = {};
+    for (int ITSlayerId = 0; ITSlayerId < 7; ITSlayerId++) {
+      if ((softPiItsHitMap & (1 << ITSlayerId)) > 0) {
```

Comment:
```suggestion
      if (TESTBIT(softPiItsHitMap, ITSlayerId)) {
```
I think you can replace this with `TESTBIT`.

Reply:
Done